### PR TITLE
able to update username, email and password

### PR DESCRIPTION
**Pull request by fnoori**

### CODE REVIEW

**Comment by nxtman123** [comment]
In your comments describing the arguments for the the event/function, make sure to mention all required or possible arguments including `confirm_email`, etc.

Before checking all the details about one of the fields, make sure to test if the new value is different from the old value. If it is the same, silently skip it. This will prevent a possible error when it is found that a user (this user, in fact) already has that email/username.

Also, check that the username and email are not too long for their db fields.

File: src/server/authentication.js [hunk 1]
```diff
@@ -87,6 +92,118 @@ module.exports = psql => ({
...
+      (optional)username, (optional) confirm_username,
+      (optional)password,  (optional) confirm_password
+    }
+  */
+  // user = { decoded token }
+  updateUserData: async (updateData, user) => {
+    try {
+      const toUpdate = {};
+      const userInDatabase = await psql('users')
+        .where('user_id', '=', user.user_id);
```
Comment (nxtman123) [inline src/server/authentication.js]: Append .first() to just get the one row outside of an array (and avoid [0] everywhere)

File: src/server/authentication.js [hunk 2]
```diff
@@ -87,6 +92,118 @@ module.exports = psql => ({
...
+
+      if (updateData.email) {
+        if (updateData.email.length > EMAIL_MAX) {
```
Comment (nxtman123) [inline src/server/authentication.js:110]: This test could be after-inside the test for if this is a real change

**Review by nxtman123 — COMMENTED** [review]
Just a couple nit-picks 😛 merge when you're ready

File: src/server/authentication.js (hunk 2 above)
Comment (fnoori) [reply src/server/authentication.js:110]: If I were to put it inside the test to see if it is a real change, I would need to potentially query the db three times, once for each check: email, username, and password.

The way I have it right now, it just needs to query the db once (drawback being, it queries even if the changes are not real).

Comment (nxtman123) [reply src/server/authentication.js:110]: I had not thought of that. Good call

**Pull request merged**